libs/ha/ha_common_libs_k8s.py

```diff
@@ -1777,15 +1783,35 @@ def iam_bucket_cruds(self, event, s3_obj=None, user_crud=False, num_users=None,
                 try:
                     LOGGER.debug("Creating %s user", i_i)
                     user = None
-                    user = self.mgnt_ops.create_account_users(nusers=1)
+                    if not header:
+                        user = self.mgnt_ops.create_account_users(nusers=1)
+                    else:
+                        payload = {}
+                        name = f"ha_iam_{i_i}_{time.perf_counter_ns()}"
+                        payload.update({"uid": name})
+                        payload.update({"display_name": name})
+                        LOGGER.info("Creating IAM user request....")
+                        endpoint = CSM_REST_CFG["s3_iam_user_endpoint"]
+                        resp = self.restapi.rest_call("post", endpoint=endpoint, json_dict=payload,
+                                                      headers=header)
+                        LOGGER.info("IAM user request successfully sent...")
+                        if resp.status_code == HTTPStatus.CREATED:
+                            resp = resp.json()
+                            user = dict()
+                            user.update({resp["keys"][0]["user"]: {
+                                "user_name": resp["keys"][0]["user"],
+                                "password": S3_CFG["CliConfig"]["s3_account"]["password"],
+                                "accesskey": resp["keys"][0]["access_key"],
+                                "secretkey": resp["keys"][0]["secret_key"]}})
```

Comment:
Instead of adding this here, please create different method, for create and delete IAM user when header is provided.